x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java

```diff
@@ -236,9 +231,8 @@ public void setUpMocks() {
         auditor = MockTransformAuditor.createMockAuditor();
         transformConfigManager = new InMemoryTransformConfigManager();
         client = new NoOpClient(getTestName());
-        // we need "generic" as part of the name, because it is asserted
-        executorName = ThreadPool.Names.GENERIC + "-" + getTestName();
-        threadPool = new TestThreadPool(executorName, new ScalingExecutorBuilder(executorName, 4, 10, TimeValue.timeValueSeconds(30)));
+        threadPool = new TestThreadPool(ThreadPool.Names.GENERIC,
+            new ScalingExecutorBuilder(ThreadPool.Names.GENERIC, 4, 10, TimeValue.timeValueSeconds(30)));
```

Comment:
I am not sure about this change. This got recently introduced:
https://github.com/elastic/elasticsearch/pull/65721

to fix:

https://github.com/elastic/elasticsearch/issues/65542

The change reverts #65721 and we might get test failures again. However, I am not sure the test problem justifies keeping the executor name. :thinking:

Reply:
I don't think that #65542 is caused by the thread pool. If an assertion fails in the test, the indexer is not stopped and continues to fire request asynchronously. So #65721 fixes the error message caused by the test failure but not the failure itself. I guess there is a race condition that can make this [`assertTrue`](https://github.com/elastic/elasticsearch/blob/d3aef910c14fbf4a8b54ebc01641ebf3feb6476d/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java#L504) to fail in some exceptional case. 
I can add a try/finally block to ensure that we stop the indexer when an assertion fails, that will clarify things the next time a failure happens. Does that make sense ?

Reply:
#65542 failed when a new search request got scheduled but the thread pool was closed. The order of timestamps show that it 1st tried to queue the request _than_ the failure happened, not vice versa. So the threadpool got closed.

However, I am fine with an alternative implementation and enhanced error handling and logging in case of the error. This should help to find the root cause the next time it happens (which seems to be rare).